src/config/stylegan.py

```diff
@@ -0,0 +1,116 @@
+import torch
+from loguru import logger as info_logger
+from src.disk import disk
+from pathlib import Path
+from src.logger.simple import Logger
+from src.data.baseline import BaselineDataset
+from src.utils.download import download_dataset
+from src.models.stylegan import StyleBased_Generator
+from src.training.stylegan import StyleGanTrainer
+from src.storage.simple import Storage
+from src.losses.gram import VGGGramLoss
+from src.losses.STRFL import OCRLoss
+from torchvision import models
+from torchvision.models.resnet import BasicBlock
+from torch.utils.data import DataLoader
+
+
+class ContentResnet(models.ResNet):
+    def _forward_impl(self, x):
+        # See note [TorchScript super()]
+        x = self.conv1(x)
+        x = self.bn1(x)
+        x = self.relu(x)
+        x = self.maxpool(x)
+
+        x = self.layer1(x)
+        x = self.layer2(x)
+        x = self.layer3(x)
+        x = self.layer4(x)
+
+        #x = self.avgpool(x)
+        #x = torch.flatten(x, 1)
+        #x = self.fc(x)
+
+        return x
+
+
+class Config:
+    def __init__(self):
+        disk.login()
+
+        device = 'cuda:0' if torch.cuda.is_available() else 'cpu'
+        info_logger.info(f'Using device: {device}')
+        style_dir = Path('data/IMGUR5K')
+        download_dataset('IMGUR5K')
+        batch_size = 16
+        train_dataloader = DataLoader(BaselineDataset(style_dir / 'train'), shuffle=True, batch_size=batch_size)
+        val_dataloader = DataLoader(BaselineDataset(style_dir / 'val'), batch_size=batch_size)
+
+        total_epochs = 500
+
+        model = StyleBased_Generator(dim_latent=512)
+        model.load_state_dict(torch.load('models/Stylegan_content_192x64/model'))
+        model.to(device)
+
+        style_embedder = models.resnet18()
+        style_embedder.fc = torch.nn.Identity()
+        style_embedder = style_embedder.to(device)
+        style_embedder.load_state_dict(torch.load('models/Stylegan_content/style_embedder'))
```

Comment:
I suppose there should be `Stylegan_content_192x64` instead of `Stylegan_content`

Reply:
also I suppose we can have something similar to  `weights_folder = 'models/Stylegan_content_192x64'` and then use it like this:
```Python
style_embedder.load_state_dict(torch.load(f'{weights_folder}/style_embedder')) 
```